src/components/sync-connected-sites.tsx

```diff
@@ -54,13 +63,41 @@ export const convertTreeToOptionsToSync = ( tree: TreeNode[] ): SyncOption[] =>
 		const wpContent = filesAndFolders.find( ( node ) => node.id === 'wp-content' )?.children || [];
 
 		wpContent.forEach( ( item ) => {
-			if ( item.checked && isSyncOption( item.id ) ) {
-				optionsToSync.push( item.id );
+			if ( isSyncOption( item.id ) ) {
+				const hasCheckedChildren = item.children?.some( ( child ) => child.checked );
+				const shouldInclude = item.checked || hasCheckedChildren;
+
+				if ( shouldInclude ) {
+					optionsToSync.push( item.id );
+
+					if ( item.id === SYNC_OPTIONS.plugins && item.children ) {
+						const selectedPlugins = item.children
+							.filter( ( plugin ) => plugin.checked )
+							.map( ( plugin ) => plugin.id );
+						if ( selectedPlugins.length > 0 && selectedPlugins.length < item.children.length ) {
+							specificSelections.plugins = selectedPlugins;
+						}
+					}
+
+					if ( item.id === SYNC_OPTIONS.themes && item.children ) {
+						const selectedThemes = item.children
+							.filter( ( theme ) => theme.checked )
+							.map( ( theme ) => theme.id );
+						if ( selectedThemes.length > 0 && selectedThemes.length < item.children.length ) {
+							specificSelections.themes = selectedThemes;
+						}
+					}
+				}
 			}
```

Comment:
I think we can remove double-conditions, code-duplications and simplify it with:
```suggestion
			if ( ! isSyncOption( item.id ) ) {
				return;
			}

			if ( item.checked ) {
				optionsToSync.push( item.id );
				return;
			}

			const checkedChildren = item.children
				?.filter( ( child ) => child.checked )
				.map( ( plugin ) => plugin.id );
				
			if ( checkedChildren?.length ) {
				if ( item.id === SYNC_OPTIONS.plugins  ) {
					specificSelections.plugins = checkedChildren;
				} else if ( item.id === SYNC_OPTIONS.themes ) {
					specificSelections.themes = checkedChildren;
				}
			}
```

Reply:
I've made some adjustments to the code to follow your suggestion.
Let me know what you think.